Bug OCPBUGS#9139: Document limitations of bare metal workers with OSP (#57798)
openshift/openshift-docs: modules/installation-osp-deploying-bare-metal-machines.adoc

Bare-metal compute machines are not supported on clusters that use Kuryr.

[WARNING]
====

> On {rh-openstack} 16.2, clusters that use bare metal workers and the OpenShift SDN can fail depending on the MTU configuration that is described in a link:https://bugzilla.redhat.com/show_bug.cgi?id=2153801[known issue].

Author: This issue is on_qa and it's likely to be verified soon.

Reviewer: How soon? Might it make sense to put a hold on this change?

Author: Sorry for the delay. Given the bug was already verified I will remove it from this docs PR
https://bugzilla.redhat.com/show_bug.cgi?id=2177155

Bare-metal compute machines are not supported on clusters that use Kuryr.

[WARNING]
====

> When installing a OpenShift cluster with bare metal workers on {rh-openstack} 16.1.7 or later releases with OVN, the cluster is unable to the create worker nodes due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue]. The link:https://bugzilla.redhat.com/show_bug.cgi?id=2083120[same issue] has been fixed on 16.2, so it's recommended to upgrade to that version to use bare metal workers.
> Also, on {rh-openstack} 16.2 OpenShift installation with bare metal workers using `OpenShiftSDN` might fail depending on the MTU used as decribed in link: https://bugzilla.redhat.com/show_bug.cgi?id=2153801[the bug]. A solution is expected once link:https://bugzilla.redhat.com/show_bug.cgi?id=2177155[this issue] is fixed.

Reviewer: ```suggestion
Bare metal workers are not functional on clusters that run on {rh-openstack} versions before 16.2 due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue].

On {rh-openstack} 16.2, clusters that use bare metal workers and the OpenShift SDN can fail depending on the MTU configuration that is described in a link:https://bugzilla.redhat.com/show_bug.cgi?id=2153801[known issue].
```

Bare-metal compute machines are not supported on clusters that use Kuryr.

[WARNING]
====

> Bare metal workers are not functional on clusters that run on {rh-openstack} versions after 16.1.6 and before 16.2.4 due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue].
> 
> On {rh-openstack} 16.2, clusters that use bare metal workers and the OpenShift SDN can fail depending on the MTU configuration that is described in a link:https://bugzilla.redhat.com/show_bug.cgi?id=2153801[known issue].

Reviewer: To be clear, we try to avoid linking to live bugs in the docs, as maintenance gets confusing. Maybe we could link those bugs to tickets to edit/remove the text in this PR if they're resolved? 

lgtm, otherwise. Could consider:

```suggestion
Bare metal workers are not functional on clusters that run on {rh-openstack} versions between 16.1.6 and 16.2.4 due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue].

On {rh-openstack} 16.2, clusters that use bare metal workers and the OpenShift SDN can fail depending on the MTU configuration that is described in a link:https://bugzilla.redhat.com/show_bug.cgi?id=2153801[known issue].

```

Author: ah okay, I will add that. Thanks for the hint

Author: I just added at https://bugzilla.redhat.com/show_bug.cgi?id=2177155

[NOTE]
====
Be sure that your `install-config.yaml` file reflects whether the {rh-openstack} network that you use for bare metal workers supports floating IP addresses or not.

> When installing a OpenShift cluster with bare metal workers on {rh-openstack} 16.1 with OVN, the cluster is unable to the create worker nodes due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue]. The link:https://bugzilla.redhat.com/show_bug.cgi?id=2083120[same issue] has been fixed on 16.2, so it's recommended to upgrade to that version to use bare metal workers.

Reviewer 2: > When installing a OpenShift cluster with bare metal workers on {rh-openstack} 16.1 with OVN, the cluster is unable to the create worker nodes due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue].

Note that the bare-metal workers setup worked up to OSP version [16.1.6](http://rhos-qe-mirror-tlv.usersys.redhat.com/rcm-guest/puddles/OpenStack/16.1-RHEL-8/z6/COMPOSE_ID) (OVN version ovn2.13-20.12.0-104.el8fdp.x86_64), and the regression is since OSP version 16.1.7 (OVN version ovn2.13-20.12.0-192.el8fdp.x86_64) - BMWs are not able to access the metadata service when running OSP 16.1.7 version or higher.

* Bare metal is available as link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.1/html/bare_metal_provisioning/configuring-the-bare-metal-provisioning-service-after-deployment#creating-the-bare-metal-flavor_bare-metal-post-deployment[a {rh-openstack} flavor].

> * Bare metal workers are not functional on clusters that run on {rh-openstack} versions between 16.1.6 and 16.2.4 due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue] that causes meta-data service to be unavailable for Services within OpenShift nodes.

Reviewer: ```suggestion
* Bare metal workers do not function on clusters that run on {rh-openstack} versions between 16.1.6 and 16.2.4 due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue] that causes the metadata service to be unavailable for services on {product-title} nodes.

```

> * Bare metal workers do not function on clusters that run on {rh-openstack} versions between 16.1.6 and 16.2.4 due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue] that causes the metadata service to be unavailable for services on {product-title} nodes.

Reviewer 3: Suggestion: Is it possible to start with a sentence that provides an explicit condition for users to meet, such as "Your {rh-openstack} version is either less than 16.1.6 or later than 16.2.4" and then follow with this added text for context? 

Either way I think users would get the message to avoid those problematic versions, but being more explicit might be a nice touch.

Reviewer: ```suggestion
* If your cluster runs on an {rh-openstack} version that is more than 16.1.6 and less than 16.2.4, bare metal workers do not function due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue] that causes the metadata service to be unavailable for services on {product-title} nodes.

```

* Bare metal is available as link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.1/html/bare_metal_provisioning/configuring-the-bare-metal-provisioning-service-after-deployment#creating-the-bare-metal-flavor_bare-metal-post-deployment[a {rh-openstack} flavor].

* Bare metal workers do not function on clusters that run on {rh-openstack} versions between 16.1.6 and 16.2.4 due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue] that causes the metadata service to be unavailable for services on {product-title} nodes.

> 

Reviewer: ```suggestion

```

* Bare metal is available as link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.1/html/bare_metal_provisioning/configuring-the-bare-metal-provisioning-service-after-deployment#creating-the-bare-metal-flavor_bare-metal-post-deployment[a {rh-openstack} flavor].

> * The {rh-openstack} does not have versions after 16.1.6 and before 16.2.4. Bare metal workers are not functional on clusters that run on those versions due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue].

Reviewer: ```suggestion
* {rh-openstack} does not have versions after 16.1.6 and before 16.2.4. Bare metal workers are not functional on clusters that run on those versions due to a link:https://bugzilla.redhat.com/show_bug.cgi?id=2033953[known issue].

```
